### ISSUE TITLE
Numba issues with RVecs

### ISSUE THREAD

**Issue by toicca** [root-project/root#12234]
### Describe the bug
When using RVecs (namely RvecF) with Numba in PyRoot the exception "Failed to extract template argument of type RVecF" raises. Changing RVecF to RVec\<float\> fixes the issue.

Numba functions also don't seem to allow defining RVecs inside them, giving 
```
TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Unknown attribute 'RVec' of type Module(<module 'ROOT' from '/cvmfs/sft.cern.ch/lcg/views/LCG_102b_swan/x86_64-centos7-gcc11-opt/lib/ROOT/__init__.py'>)

File "../../../../../../tmp/ipykernel_594/4272152336.py", line 3:
<source missing, REPL/exec in use?>

During: typing of get attribute at /tmp/ipykernel_594/4272152336.py (3)

File "../../../../../../tmp/ipykernel_594/4272152336.py", line 3:
<source missing, REPL/exec in use?>
```

when defining a function such as

```
@ROOT.Numba.Declare(['RVec<float>', 'int'], 'RVec<float>')
def pypowarray(x, y):
    v = ROOT.RVec['float']()
    return x**y
```

### To Reproduce
The PyRoot tutorial pyroot004_NumbaDeclare.py (https://root.cern.ch/doc/master/pyroot004__NumbaDeclare_8py.html) directly reproduces the issue with RVecF. Modifying the code where pypowarray is defined to match the code above reproduces the issue with defintions.

### Setup
1. ROOT Version: 6.26/08
2. Operating system: CentOS 7
3. I used SWAN notebooks with LCG 102b. The notebook was created directly from the "Open in SWAN" button in the tutorial.

**Comment by eguiraud** [comment]
Hi @toicca , thank you for the report!

About `RVecF` in the signature not working while `RVec<float>` does, I think that was reported in https://github.com/root-project/root/issues/11294 and it should be fixed by https://github.com/root-project/root/pull/11553 . That PR is waiting on a review from me, I'll get to it as soon as possible.

About `numba` not being able to interpret a Python function that uses `ROOT.RVec`: that's technically a limitation of numba, and [there is some work](https://indico.cern.ch/event/1106990/contributions/4991292/) towards having numba understand PyROOT-wrapped C++, but for now "PyROOT things" (or cppyy-wrapped C++, if you want) is not part of the subset of Python that numba can deal with.

**Comment by toicca** [comment]
Hi @eguiraud, thanks for the response!

Since the `RVecF` is being addressed and `numba` just isn't currently built to work with PyROOT stuff I think this issue can be closed.

Lastly, just to clarify, there isn't a way for me to use `numba` with `RDF` to return an array that I define inside the function? For example, if I had the code below there isn't any type `'array'` that I could use?

```
@ROOT.Numba.Declare(['int', 'float'], 'array')
def pyfillarray(N, x):
    array = ...
    for i in range(N):
        array[i] = x
    return array
```

**Comment by eguiraud** [comment]
`Numba.Declare` converts returned numpy arrays to RVecs, so this should work:

```
@ROOT.Numba.Declare(['int', 'float'], 'RVec<float>')
def pyfillarray(N, x):
    array = np.array(..., dtype=np.float32)
    for i in range(N):
        array[i] = x
    return array
```
